src/components/dashboard/Dashboard.js

```diff
@@ -190,32 +190,40 @@ const Dashboard = props => {
   }, [navigation, showDeleteAccountDialog])
 
   const getFeedPage = useCallback(
-    async (reset = false) => {
-      const res =
-        (await userStorage
+    debounce(
+      async (reset = false) => {
+        log.debug('getFeedPage:', { feeds, loadAnimShown, didmount })
+        const feedPromise = userStorage
           .getFormattedEvents(PAGE_SIZE, reset)
-          .catch(e => logger.error('getInitialFeed -> ', e.message, e))) || []
-
-      if (res.length === 0) {
-        log.warn('empty feed')
-        return
-      }
-
-      if (reset) {
-        // a flag used to show feed load animation only at the first app loading
-        if (!loadAnimShown) {
-          // a time to perform feed load animation till the end
-          await delay(1900)
-          store.set('feedLoadAnimShown')(true)
+          .catch(e => logger.error('getInitialFeed -> ', e.message, e))
+
+        if (reset) {
+          // a flag used to show feed load animation only at the first app loading
+          //subscribeToFeed calls this method on mount effect without dependencies because currently we dont want it re-subscribe
+          //so we use a global variable
+          if (!didmount) {
```

Comment:
1. you could use a ref initialized with store.get('feedLoadAnimShown') instead of global variable
2. according to the logic `didmount` means 'some anomation was shown and feed is loaded'. It would be good to rename this variable as 'didmount' refers to the component lifecycle - 'after first render' which isn't consistent 
3. why you're waiting for animation completes ? If feed will be loaded in 0.5sec, we'll still see the animation the 1.5 seconds left. From my point of view better to cancel animation once feed is loaded.
4. An opposite case - what if feed will load longer that 2sec ? Animation will disappear and we'll see empty screen ?  
I think we'll need to use cycled animation and don't await for any animation event:
```
startAnimation()
loadFeed().finally(() => immediatelyStopAnimation())
```

Reply:
1. it doesnt work with the gun callback
2. i agree
3. yes. that was a requirement from the designer
4. it shows the 3 blank items.